contracts/eosio.system/delegate_bandwidth.cpp

```diff
@@ -97,10 +97,14 @@ namespace eosiosystem {
                             /*  no need to parse authorites
                             const authority& owner,
                             const authority& active*/ ) {
-      auto name_str = eosio::name{newact}.to_string();
 
-      eosio_assert( name_str.size() == 12 || creator == N(eosio), "account names must be 12 chars long" );
-      eosio_assert( name_str.find_first_of('.') == std::string::npos  || creator == N(eosio), "account names cannot contain '.' character");
+      if( creator != _self ) {
+         auto tmp = newact;
+         for( uint32_t i = 0; i < 12; ++i ) {
+           if( tmp & 0x1f ) eosio_assert( false, "name may not contain '.'"); 
```

Comment:
This will detect name's with less than 12 characters, but it will not give the users any idea what to fix since they will have no idea why their name was rejected for containing a '.' when they didn't put one in there.

Reply:
it is actually wrong everything is shifted 4 bits in the wrong direction.